src/Transform/LowerKrnlShape.cpp

```diff
@@ -45,18 +45,28 @@ class LowerKrnlShape : public OpRewritePattern<KrnlShapeOp> {
     auto rank =
         convertToMemRefType(krnlShapeOp.alloc().getType()).getShape().size();
 
+    // Create MemRef to hold shape information.
+    SmallVector<int64_t, 1> memRefShape;
+    memRefShape.emplace_back(rank);
+    auto memRefType = MemRefType::get(memRefShape, rewriter.getIndexType());
+    auto newMemRefAlloc = rewriter.create<AllocOp>(loc, memRefType);
+
     SmallVector<mlir::Value, 4> fromExtentsOpOperands;
     for (int idx = 0; idx < rank; idx++) {
       auto index = rewriter.create<ConstantOp>(
           loc, rewriter.getIntegerAttr(rewriter.getIndexType(), idx));
       auto operand = rewriter.create<KrnlDimOp>(
           loc, rewriter.getIndexType(), krnlShapeOp.alloc(), index);
       fromExtentsOpOperands.emplace_back(operand);
+
+      // Store value in the new MemRef.
+      auto idxValue =
+          emitConstantOp(rewriter, loc, rewriter.getIndexType(), idx);
+      SmallVector<Value, 1> indexArg = {idxValue};
+      rewriter.create<StoreOp>(loc, operand, newMemRefAlloc, indexArg);
```

Comment:
Would prefer to use AffineStoreOp here.

Reply:
Any reasons not to use affine Store? You mark the issue as resolved but you did not make it affine. Please explain.

Reply:
Ah thought I did this one :) Thanks! It's done now!